breadbox/breadbox/crud/dataset.py

```diff
@@ -108,14 +109,28 @@ def get_datasets(
     dataset_poly = with_polymorphic(Dataset, [MatrixDataset, TabularDataset])
 
     filter_clauses = [Dataset.group_id.in_(group_ids)]  # pyright: ignore
+
     # Don't return transient datasets
     filter_clauses.append(Dataset.is_transient == False)
-    # TODO: Below filters only returns for matrix datasets!
+    # TODO: 'feature_id' and 'sample_id' filters only returns for matrix datasets!
     # Decide if should return for metadata when given feature id/type or sample id/type
+    # TODO: feature type can be none. How should we filter those datasets?
     if feature_type is not None:
-        filter_clauses.append(
-            dataset_poly.MatrixDataset.feature_type_name == feature_type
+        # Make sure that the `feature_type` dimension type is actually in the feature axis so tabular datasets are correctly filtered
+        feature_dimension_type = (
+            db.query(DimensionType).filter_by(name=feature_type).one_or_none()
         )
+        if feature_dimension_type and feature_dimension_type.axis == "feature":
+            filter_clauses.append(
+                or_(
+                    dataset_poly.MatrixDataset.feature_type_name == feature_type,
+                    dataset_poly.TabularDataset.index_type_name == feature_type,
+                )
+            )
+        else:
+            filter_clauses.append(
+                dataset_poly.MatrixDataset.feature_type_name == feature_type
+            )
```

Comment:
This filter clause has been modified so that a dimension type with 'sample' as the axis doesn't get used as a feature_type filter